### PR TITLE
Update Azure auth docs

### LINKED ISSUE #5044: Docs: Update Azure documentation to reflect ADLS Gen2 support
- Added support for HNS
- Using NewDefaultAzureCredential in case access key not provided which checks for:
  - Env variables
  - MSI
  - CLI credentials

### PR DESCRIPTION
**Pull request by itaiad200**
closes #5044 
Closes ##5125

### CODE REVIEW

**Review by N-o-Z — DISMISSED** [review]
Thanks,
We are missing a section that describes all the azure storage types we support - specifically
Azure Blob Storage
Azure Data Lake Gen2

Also maybe add something about support for multi account access / operations?

File: docs/deploy/azure.md [hunk 1]
```diff
@@ -19,11 +19,90 @@ next:  ["Import data into your installation", "../howto/import.html"]
...
+1. Import of data from different storage accounts
+2. Copy/Read/Write of data that was imported from a different storage account
+3. Create pre-signed URL for data that was imported from a different storage account
+
+### Identity Based Credentials
+
+lakeFS uses environment variables to determine credentials to use for authentication. the following authentication methods are supported:
+1. Managed Service Identity (MSI)
+2. Service Principal RBAC
+3. Azure CLI
```
Comment (itaiad200) [inline docs/deploy/azure.md:56]: It this a tool to authenticate with or another type of authentication like MSI? I know that the Azure docs are confusing, I don't want to import that confusion here as well

File: docs/deploy/azure.md [hunk 2]
```diff
@@ -19,11 +19,90 @@ next:  ["Import data into your installation", "../howto/import.html"]
...
+
+
+**Note:** It is possible to provide both account based credentials and environment variables to lakeFS. In that case - lakeFS will use
+the account credentials for any access to data located in the given account, and will try to use the identity credentials for any data located outside the given account.
+{: .note }
+
 ## Create a database
 
-lakeFS requires a PostgreSQL database to synchronize actions in your repositories.
+lakeFS stores metadata in a database for its versioning engine. This is done via a Key-Value interface that can be implemented on any DB engine and lakeFS comes with several built-in driver implementations (You can read more about it [here](https://docs.lakefs.io/understand/how/kv.html)).
```
Comment (itaiad200) [inline docs/deploy/azure.md:103]: Nice catch!

**Review by itaiad200 — COMMENTED** [review]
Great explanations! I wish I could Approve this but somehow I'm the author......

File: docs/deploy/azure.md (hunk 1 above)
Comment (N-o-Z) [reply docs/deploy/azure.md:56]: No tool - just different env vars for different methods of authentication

**Comment by N-o-Z** [comment]
> Great explanations! I wish I could Approve this but somehow I'm the author......

@itaiad200  With your permission - I will approve the PR on your behalf 😄

**Review by N-o-Z — APPROVED** [review]
@itaiad200 says: "I approve" 😄

**Pull request merged**